renku/core/management/clone.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2019- Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Clone a Renku repo along with all Renku-specific initializations."""
+
+import os
+
+from git import GitCommandError, Repo
+
+from renku.core import errors
+from renku.core.management.githooks import install
+
+
+def clone(
+    url,
+    path=None,
+    install_githooks=True,
+    install_lfs=True,
+    skip_smudge=True,
+    progress=None
+):
```

Comment:
Can we add an option for a shallow clone?

Reply:
Do you have a use-case for it? Here, I believe we always need the whole Renku project history and shallow won't be used. Moreover, shallow cloning has many flags (e.g. `--depth`, `--shallow-since`, `--shallow-exclude`, ...) and I don't want to include them all here if not needed. However, if you have a use for it probably we can include `--depth`.

Reply:
Yes, most likely we will have to expose this on the service, cause for a simple use case like adding a file from UI we probably don't want full clone (cause full clone on big repositories could be quite slow). I think adding a `--depth` would be sufficient for that use case. We can also make a ticket for that if you feel it's out of scope here.

Reply:
No, it's OK and already done 👍